structure.go

```diff
@@ -72,16 +72,16 @@ func NewStructureRef(path string) *Structure {
 
 // DropTransientValues removes values that cannot be recorded when the
 // dataset is rendered immutable, usually by storing it in a cafs
-func (st *Structure) DropTransientValues() {
-	st.Path = ""
+func (s *Structure) DropTransientValues() {
```

Comment:
nice! I think we can also turn the golint check in circleci back on now that `dataset.DatasetPod` is gone. If we don't land that in this PR I'll happily follow up later.